Add support for specifying CA cert

alexg-axis opened the pull request:
Solves #188

alexg-axis committed:
Add support for specifying CA cert

EdJoPaTo commented on src/cli.rs:
```diff
@@ -243,6 +243,20 @@ pub struct MqttConnection {
+        value_name = "FILEPATH",
+        global = true,
+    )]
+    pub ca_cert: Option<std::path::PathBuf>,
```
Not sure about the name here. CA Certificate Path. value name already suggests path so not including that is a good idea. CA ist certificate authority. Sometimes this variable is called ca_cert and sometimes ca_certificate_path. Certificate authority certificate. Maybe authority_certificate is the most readable one?

Also I think, having the same variable name in all places is probably a good idea.

EdJoPaTo reviewed (commented):
Thank you for providing a PR for this!

I have some variable naming / argument naming thoughts to simplify understanding the args / code. But other than that it looks good!

EdJoPaTo committed:
refactor(encryption): improve variable names

Keep them similar in different places.
_path is superfluous as their type is already &Path.

EdJoPaTo committed:
fix(encryption): add error contexts

As there are multiple different certificates handled now its harder to
figure out which certificate created an error.

EdJoPaTo committed:
Merge remote-tracking branch 'origin/main' into dev/ca-certs

Conflicts:
	src/cli.rs
	src/mqtt/connect.rs
	src/mqtt/encryption.rs

EdJoPaTo commented:
Following https://test.mosquitto.org/ssl/ ends up with `invalid peer certificate: Other(OtherError(UnsupportedCertVersion))`. rustls does not work with all the formats out there, is that the case here too? Or have I done anything wrong? Or is there some issue elsewhere?
